### ISSUE TITLE
Incorrect `incompatible type` error in `@overload`

### ISSUE THREAD

**Issue by pvandyken** [microsoft/pyright#9136]
This issue is possibly the same as that reported in #9054. I opened a new issue because I've traced mine back to a different version and the example is somewhat different.

**Describe the bug**
Function overload previously allowed as compatible are now marked as incompatible as of version 1.1.378. I did not notice anything in the changelog that might explain this. 

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
from typing import Literal, overload

# obviously this type of magic keyword is pretty inadvisable,
# but it makes more sense in the code it's drawn from
@overload
def test(
    return_str: Literal[True] | None = ...,
) -> str: ...
@overload
def test(
    return_str: Literal[False] = ...,
) -> int: ...
```

From the command line:

```
/home/.../test.pyi
  /home/.../test.pyi:4:5 - error: Overload 1 for "test" overlaps overload 2 and returns an incompatible type (reportOverlappingOverload)
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**
Both the extension and CLI.

**Comment by erictraut** [comment]
Pyright is working as intended here. The change in behavior is due to a bug fix in 1.1.378. It's the last bullet under "bug fixes" in the [release notes](https://github.com/microsoft/pyright/releases/tag/1.1.378).

The issue here is that if someone calls `test()` with no arguments, bother overloads 1 and 2 apply, but they have incompatible return types. I suspect what you intend is for the second overload to have no default argument. If you delete the `= ...` in overload 2, the error goes away.

**Comment by pvandyken** [comment]
Ah, I understand, apologies for the false alarm.
